### ISSUE TITLE
ForeignKey inference for derived classes not working properly

### ISSUE THREAD

**Issue by RobinBuschmann** [RobinBuschmann/sequelize-typescript#395]
Given following models:
```ts
@Table
class Related extends Model<Related> {
  @ForeignKey(() => Parent) parentId: number;
  @BelongsTo(() => Parent) parent: Parent;
}
```
```ts
@Table
class Parent extends Model<Parent> {
  @HasMany(() => Related) relatedOnes: Related;
}
```
```ts
@Table
class Derived extends Parent {}
```
```ts
const sequelize = new Sequelize({...})
sequelize.addModels([Related,  Parent, Derived]);
```

Expected behaviour:
Should infer foreign key for `Child` on `Related`

Actual behaviour:
Throws `Foreign key for "Derived" is missing on "Related".`

**Comment by chrift** [comment]
@RobinBuschmann Did you ever get to the bottom of this? Or find a workaround? 
I think I have all my references defined correctly but am still getting this error and don't know what to do!

**Comment by Queatz** [comment]
We're seeing this as well

**Comment by RobinBuschmann** [comment]
@chrift sry for the late reply - I've totally overseen your comment. A workaround is, explicitly setting the foreign keys in the HasMany/BelongsTo/HasOne/BelongsToMany decorators like so: `@HasMany(Related, 'parentId')` and `@BelongsTo(Parent, 'parentId')`.

**Comment by RobinBuschmann** [comment]
Fixed with `sequelize-typescript@1.0.0`

**Comment by Arlodotexe** [comment]
I'm seeing this problem again
